packages/@aws-cdk/aws-apigateway/README.md

```diff
@@ -845,6 +845,46 @@ By performing this association, we can invoke the API gateway using the followin
 https://{rest-api-id}-{vpce-id}.execute-api.{region}.amazonaws.com/{stage}
 ```
 
+## Gateway response
+To add specific paramters or change the api response you can configure gateway response at the api level.
+A gateway response is indentified by response type defined by API gateway.
+For supported event types refer to [here](https://docs.aws.amazon.com/apigateway/latest/developerguide/supported-gateway-response-types.html)
+
+Here is an example to configure gateway response for supported response type:
+
+```ts
+const api = new apigateway.RestApi(this, 'books-api');
+api.addGatewayResponse('test-response', {
+      type: ResponseType.ACCESS_DENIED,
+      statusCode: '500',
+      parameters: {
+        'gatewayresponse.header.Access-Control-Allow-Origin': "test.com",
+        'gatewayresponse.header.test-key': 'test-value'
+      },
+      templates: {
+        'application/json': "{ \"message\": $context.error.messageString, \"statusCode\": '488', \"type\": \"$context.error.responseType\" }"
+      }
+});
+```
+
+You can also initialized a new gateway response for an exixting api as below :
+
+```ts
+const api = apigateway.RestApi.fromRestApiId(this, 'books-api', 'abhb8fds');
+new GatewayResponse(this, id, {
+      restApi: api,
+      type: ResponseType.ACCESS_DENIED,
+      statusCode: '500',
+      parameters: {
+        'gatewayresponse.header.Access-Control-Allow-Origin': "test.com",
+        'gatewayresponse.header.test-key': 'test-value'
+      },
```

Comment:
This looks cumbersome and error prone. We could make it better by lifting some of the human error out.

Perhaps something like -

```ts
responseParameters: {
  headers: {
    'Access-Control-Allow-Origin': 'test.com',
  },
  querystring: {
    ...
  },
  paths: {
    ...
  }
}
```

Reply:
you mean to say in in gateway response options we have property like below 
readonly parameters?: ResponseParameters;
And inside GatewayResponse constructor we convert the above values to required format ?

export interface ResponseParameters {
  headers?: {[key: string] : string}
  querystring?: {[key: string] : string}
  paths?: {[key: string] : string}
}

Reply:
Yes, something along those lines.

Reply:
On the other hand, I'm wondering if querystring and path make sense here. 

As far as I can see from documentation on HTTP, HTTP responses have headers but they don't have paths or query strings. Am I missing something?

What does it mean when someone specifies a path key-value in `responseParameters` property of `GatewayResponse`?

Reply:
I am also not aware about how querystring or path will be returned in HTTP response.
On below link i can see gateway response can have header,querystring and path values :
[https://docs.aws.amazon.com/apigateway/latest/developerguide/api-gateway-swagger-extensions-gateway-responses.responseParameters.html](url)

Reply:
In that case, let's drop query string and path.

Let's make this a property in `GatewayResponseProps` called `responseHeaders`.